modules/nw-configure-ingress-access-logging.adoc

```diff
@@ -0,0 +1,115 @@
++
+[source, yaml]
+----
+apiVersion: operator.openshift.io/v1
+kind: IngressController
+metadata:
+  name: default
+  namespace: openshift-ingress-operator
+spec:
+  replicas: 2
+  endpointPublishingStrategy:
+    type: NodePortService <1>
+  logging:
+    access:
+      destination:
+        type: container
+----
+<1> `NodePortService` is not required to configure Ingress access logging to a sidecar. Ingress logging is compatible with any `endpointPublishingStrategy`.
++
+* When you create a sidecar container for logging, it creates a container named `logs` inside the router namespace. You can view the logs using the following command:
++
+----
+$ oc -n openshift-ingress logs  deployment.apps/router-internalapps -c logs
+----
++
+.Sample output:
+----
+2020-04-24T04:27:51.158772+00:00 router-internalapps-6bd9c8f99c-m8bhs router-internalapps-6bd9c8f99c-m8bhs haproxy[30]: Proxy public started.
+2020-04-24T04:27:51.158795+00:00 router-internalapps-6bd9c8f99c-m8bhs router-internalapps-6bd9c8f99c-m8bhs haproxy[30]: Proxy public_ssl started.
+2020-04-24T04:27:51.158800+00:00 router-internalapps-6bd9c8f99c-m8bhs router-internalapps-6bd9c8f99c-m8bhs haproxy[30]: Proxy be_sni started.
+2020-04-24T04:27:51.158814+00:00 router-internalapps-6bd9c8f99c-m8bhs router-internalapps-6bd9c8f99c-m8bhs haproxy[30]: Proxy fe_sni started.
+----
+
+Configure Ingress access logging to a syslog endpoint.
+
+* To configure Ingress access logging, you must specify a destination using `spec.logging.access.destination`. To specify logging to a syslog endpoint destination, you must specify `syslog` for `spec.logging.access.destination.type`. If the destination type is `syslog`, you must also specify a destination endpoint using `spec.logging.access.destination.syslog.endpoint` and you can specify a facility using `spec.logging.access.destination.syslog.facility`. The following example is the definition of an Ingress Controller that logs to a `syslog` destination:
++
+[source, yaml]
+----
+apiVersion: operator.openshift.io/v1
+kind: IngressController
+metadata:
+  name: default
+  namespace: openshift-ingress-operator
+spec:
+  replicas: 2
+  endpointPublishingStrategy:
+    type: NodePortService
+  logging:
+    access:
+      destination:
+      type: Syslog
+----
+
+Configure Ingress access logging with a specific log format.
+
+* You can specify `spec.logging.access.httpLogFormat` to customize the log format. The following example is the definition of an Ingress Controller that logs to a `syslog` endpoint with IP address 1.2.3.4 and port 10514:
++
+[source, yaml]
+----
+apiVersion: operator.openshift.io/v1
+kind: IngressController
+metadata:
+  name: default
+  namespace: openshift-ingress-operator
+spec:
+  replicas: 2
+  endpointPublishingStrategy:
+    type: NodePortService
+  logging:
+    access:
+      destination:
+        type: Syslog
+        syslog:
+          endpoint: 1.2.3.4:10514
```

Comment:
The endpoint needs to be specified using `address` and `port`:
```
  logging:
    access:
      destination:
        type: Syslog
        syslog:
          address: 1.2.3.4
          port: 10514
```
You might be going off an outdated version of the enhancement proposal, which has `endpoint` instead of `address` and `port`; I update the enhancement proposal to have `address` and `port` here:  https://github.com/openshift/enhancements/pull/294

Reply:
Also, the example is missing `httpLogFormat`:
```yaml
spec:
  logging:
    access:
      destination:
        type: Syslog
        syslog:
          address: 1.2.3.4
          port: 10514
      httpLogFormat: '%ci:%cp [%t] %ft %b/%s %B %bq %HM %HU %HV'
```

Reply:
You're right. I'm sorry I didn't even see the updated version!

Reply:
No worries—it's my fault for not getting the update merged.